modules/util/portal-tools-db-partition-migration-validator-test/src/testIntegration/java/com/liferay/portal/tools/db/partition/migration/validator/test/DBPartitionMigrationValidatorTest.java

```diff
@@ -0,0 +1,222 @@
+		_company = CompanyTestUtil.addCompany();
+
+		_outputDirectoryFile = new File(PropsValues.LIFERAY_HOME, "exports");
+
+		_outputDirectoryFile.mkdirs();
+	}
+
+	@After
+	public void tearDown() throws IOException, PortalException {
+		_deleteCompany();
+		FileUtils.deleteDirectory(_outputDirectoryFile);
+		System.setErr(_originalErrPrintStream);
+		System.setOut(_originalOutPrintStream);
+	}
+
+	@Test
+	public void testFailure() throws Exception {
+		String sourceFileName = _runExport(_company.getCompanyId());
+
+		String targetFilePath = _runExport(_defaultCompanyId);
+
+		File[] files = _outputDirectoryFile.listFiles();
+
+		Assert.assertEquals(Arrays.toString(files), 2, files.length);
+
+		_runValidation(
+			() -> {
+			},
+			runtimeException -> {
+				String string = _outByteArrayOutputStream.toString();
+
+				Assert.assertTrue(
+					string.contains(
+						"[ERROR] Company ID " + _company.getCompanyId() +
+							" already exists in the target database"));
+				Assert.assertTrue(
+					string.contains(
+						StringBundler.concat(
+							"[WARN] Company name ", _company.getName(),
+							" already exists in the target database. You must ",
+							"set a different value in ",
+							"DBPartitionInsertVirtualInstanceConfiguration.",
+							"config.")));
+				Assert.assertTrue(
+					string.contains(
+						StringBundler.concat(
+							"[WARN] Virtual host ",
+							_company.getVirtualHostname(),
+							" already exists in the target database. You must ",
+							"set a different value in ",
+							"DBPartitionInsertVirtualInstanceConfiguration.",
+							"config.")));
+				Assert.assertTrue(
+					string.contains(
+						StringBundler.concat(
+							"[WARN] Web ID ", _company.getWebId(),
+							" already exists in the target database. You must ",
+							"set a different value in ",
+							"DBPartitionInsertVirtualInstanceConfiguration.",
+							"config.")));
+			},
+			sourceFileName, targetFilePath);
+	}
+
+	@Test
+	public void testSuccess() throws Exception {
+		String sourceFileName = _runExport(_company.getCompanyId());
+
+		_deleteCompany();
+
+		String targetFileName = _runExport(_defaultCompanyId);
+
+		File[] files = _outputDirectoryFile.listFiles();
+
+		Assert.assertEquals(Arrays.toString(files), 2, files.length);
+
+		_runValidation(
+			() -> {
+				Assert.assertTrue(
+					_outByteArrayOutputStream.toString(
+					).isEmpty());
+				Assert.assertTrue(
+					_errByteArrayOutputStream.toString(
+					).isEmpty());
+			},
+			runtimeException -> Assert.assertEquals(
+				"0", runtimeException.getMessage()),
+			sourceFileName, targetFileName);
+	}
+
+	private void _deleteCompany() throws PortalException {
+		if (_company != null) {
+			_companyLocalService.deleteCompany(_company);
+		}
+
+		_company = null;
+	}
+
+	private String _runExport(long companyId) throws Exception {
+		try (SafeCloseable safeCloseable =
+				CompanyThreadLocal.setWithSafeCloseable(companyId)) {
+
+			return ReflectionTestUtil.invoke(
```

Comment:
Why we can not just call the main with the export option?

Reply:
We cannot call the export option directly because the tool gets the JDBC URL, user and password of the database that is going to be exported as input parameters. In integration tests I didn't find any way to get that data an input it to the program, so I had to call the inner methods by reflection.

Reply:
Got it. If the rest works. Please, send to Brian